CustomCraftSML/Serialization/Entries/CustomFood.cs

```diff
@@ -0,0 +1,298 @@
+        }
+
+        public short FoodValue
+        {
+            get => foodValue.Value;
+            set => foodValue.Value = value;
+        }
+
+        public short WaterValue
+        {
+            get => waterValue.Value;
+            set => waterValue.Value = value;
+        }
+
+        public float DecayRateMod
+        {
+            get => decayrate.Value;
+            set => decayrate.Value = value;
+        }
+
+        public bool AllowOverfill
+        {
+            get => allowOverfill.Value;
+            set => allowOverfill.Value = value;
+        }
+
+        internal bool Decomposes => this.DecayRateMod > 0f;
+
+        protected static List<EmProperty> CustomFoodProperties => new List<EmProperty>(AliasRecipeProperties)
+        {
+            new EmProperty<FoodModel>(FoodModelKey, FoodModel.None) { Optional = true },
+            new EmProperty<TechType>(SpriteItemIdKey, TechType.None) { Optional = true },
+            new EmProperty<short>(FoodKey, 0) { Optional = false },
+            new EmProperty<short>(WaterKey, 0) { Optional = false },
+            new EmProperty<float>(DecayRateKey, 0) { Optional = true },
+            new EmYesNo(OverfillKey, true) { Optional = true },
+        };
+
+        internal TechType FoodPrefab
+        {
+            get
+            {
+                if (this.FoodType == FoodModel.None)
+                {
+                    if (this.FoodValue >= this.WaterValue)
+                    {
+                        return TechType.NutrientBlock;
+                    }
+                    else
+                    {
+                        return TechType.FilteredWater;
+                    }
+                }
+
+                return (TechType)this.FoodType;
+            }
+        }
+
+        internal string IconName
+        {
+            get
+            {
+                string imagePath = IOPath.Combine(FileLocations.AssetsFolder, $"{this.ItemID}.png");
+
+                if (File.Exists(imagePath))
+                    return imagePath;
+
+                if (this.FoodValue >= this.WaterValue)
+                    return "cake.png";
+                else
+                    return "juice.png";
+            }
+        }
+
+        public CustomFood() : this(TypeName, CustomFoodProperties)
+        {
+        }
+
+        protected CustomFood(string key) : this(key, CustomFoodProperties)
+        {
+        }
+
+        protected CustomFood(string key, ICollection<EmProperty> definitions) : base(key, definitions)
+        {
+            foodValue = (EmProperty<short>)Properties[FoodKey];
+            waterValue = (EmProperty<short>)Properties[WaterKey];
+            decayrate = (EmProperty<float>)Properties[DecayRateKey];
+            allowOverfill = (EmYesNo)Properties[OverfillKey];
+
+            techGroup.Value = TechGroup.Survival;
+            techCategory.DefaultValue = TechCategory.CookedFood;
+            foodModel = (EmProperty<FoodModel>) Properties[FoodModelKey];
```

Comment:
Group the backing field assignments together and re-order them to match the same order in which they are defined.